sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -33,17 +34,21 @@ public abstract class DoNotHardcodeCredentialsBase<TSyntaxKind> : ParameterLoadi
         where TSyntaxKind : struct
     {
         protected const string DiagnosticId = "S2068";
-        protected const string MessageFormat = "Make sure hard-coded credential is safe.";
-
-        private const string DefaultCredentialWords = "password, passwd, pwd";
+        private const string MessageFormat = "{0}";
+        private const string MessageFormatCredential = @"{0} detected here, make sure this is not a hard-coded credential.";
+        private const string MessageUriUserInfo = "Review this hard-coded URL, which may contain a credential.";
+        private const string DefaultCredentialWords = "password, passwd, pwd, passphrase";
 
+        protected readonly DiagnosticDescriptor rule;
         private string credentialWords;
         private IEnumerable<string> splitCredentialWords;
         private Regex passwordValuePattern;
+        private readonly Regex uriUserInfoPattern = new Regex(@"\w+:\/\/(?<Login>[^:]+):(?<Password>[^@]+)@", RegexOptions.Compiled);
```

Comment:
I think we coukld move this regex insided the nested class, next to the existing `validCredentialPattern` regex, as it is only used there.